controllers/agentaction_controller.go

```diff
@@ -476,31 +476,41 @@ func (r *AgentActionReconciler) resolveAgentConfig(ctx context.Context, log logr
 		log.V(Log4Debug).Info("Found porter agent configuration",
 			"level", level,
 			"namespace", config.Namespace,
-			"name", config.Name)
+			"name", config.Name,
+			"plugin", config.Spec.Plugins)
 	}
 
+	var isCfgReady bool
 	// Read agent configuration defined at the system level
 	systemCfg := &porterv1.AgentConfig{}
 	err := r.Get(ctx, types.NamespacedName{Name: "default", Namespace: operatorNamespace}, systemCfg)
 	if err != nil && !apierrors.IsNotFound(err) {
-		return porterv1.AgentConfigSpec{}, errors.Wrap(err, "cannot retrieve system level porter agent configuration")
+		return porterv1.AgentConfigSpecAdapter{}, errors.Wrap(err, "cannot retrieve system level porter agent configuration")
 	}
+	isCfgReady = err == nil && systemCfg.Status.Ready
 	logConfig("system", systemCfg)
 
 	// Read agent configuration defined at the namespace level
 	nsCfg := &porterv1.AgentConfig{}
 	err = r.Get(ctx, types.NamespacedName{Name: "default", Namespace: action.Namespace}, nsCfg)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return porterv1.AgentConfigSpec{}, errors.Wrap(err, "cannot retrieve namespace level porter agent configuration")
+	if !apierrors.IsNotFound(err) {
+		if err != nil {
+			return porterv1.AgentConfigSpecAdapter{}, errors.Wrap(err, "cannot retrieve namespace level porter agent configuration")
+		}
+		isCfgReady = nsCfg.Status.Ready
 	}
 	logConfig("namespace", nsCfg)
 
 	// Read agent configuration override
```

Comment:
Just wanted to wrap my head around this. So there's the Agent Config that's set as "default" in the namespace and those values get read in then the actual agent action could potentially be using a non default named agent config which gets merged with the default namespace config? So you can define overrides in a named config and it will merge with the default config? My question is what happens if there's a plugin defined in the default config and a different plugin in the override config? Those plugin lists will merge correct? In which case will the appropriate volumes be mounted properly? Since the default config will point to one volume for its plugins and the named config will point to another. Or does the named config just override the whole "plugins" portion of the agent config and we only use that value and disregard the plugins list in the default config?